src/smc-webapp/project/file-listing/file-listing.tsx

```diff
@@ -69,9 +69,32 @@ export class FileListing extends React.Component<Props> {
     this.list_ref = React.createRef();
   }
 
+  shouldComponentUpdate(next) {
+    return (
+      !isEqual(this.props.listing, next.listing) ||
```

Comment:
I think props.listing could easily have 500 entries in it. If this is getting called on every single keystroke in a different, then project even this test will waste significant CPU.  We should fix this problem right instead, e.g., by only updating the FileListing if the actual listing changes, which can surely be controlled at a higher level or by passing in a counter instead.